numpy/_core/src/umath/ufunc_object.c

```diff
@@ -4771,6 +4771,7 @@ PyUFunc_FromFuncAndDataAndSignatureAndIdentity(PyUFuncGenericFunction *func, voi
             return NULL;
         }
     }
+    ufunc->dict = PyDict_New();
```

Comment:
One more thing:  We can glance over `tp_traverse`, but this definitely needs to be considered in `tp_dealloc`.

It would also be nice to require the ufunc setdocs code to actually use this.  Although that is better as a follow-up.

Reply:
Done! I added `Py_XDECREF` and `Py_VISIT` where needed.

Reply:
Hmm, Ok, looks like CI didn't like this change, I need to take a look at it, as I can't reproduce it on my linux x86_64 machine.

Reply:
You probably forgot the null initialization in case of an early error during creation.

Reply:
Looks like `Py_XDECREF` is causing a segfault in a few jobs (with the null check when creating the dict).

Reply:
As I said, you need to NULL initialize the slot somewhere far up in the function probably.

Reply:
Sorry! Now I get it - done!